src/messageReactionAdd.ts

```diff
@@ -11,35 +12,73 @@ const getTotalReactionCount = async (reaction: MessageReaction) => {
             try {
                 await value.fetch();
             } catch (error) {
-                return Promise.resolve(0)
+                return Promise.resolve(0);
             }
         }
 
-        return Promise.resolve(value.count!)
-    })
+        return Promise.resolve(value.count!);
+    });
 
-    const results = await Promise.all(getReactionCounts)
+    const results = await Promise.all(getReactionCounts);
 
     // reduce down the array of numbers to the count of reactions
-    return results.reduce((acc, current) => acc + current, 0)
+    return results.reduce((acc, current) => acc + current, 0);
 }
 
-export const messageReactionAdd = async (reaction: MessageReaction) => {
+export const messageReactionAdd = async (reaction: MessageReaction, user: User | PartialUser) => {
+    if (user.partial) {
+        try {
+            await user.fetch();
+        } catch (error) {
+            log.error('Something went wrong when fetching the user: ', error);
+            return;
+        }
+    }
+
     if (reaction.partial) {
         try {
             await reaction.fetch();
         } catch (error) {
-            console.error('Something went wrong when fetching the message: ', error);
+            log.error('Something went wrong when fetching the reaction: ', error);
+            return;
+        }
+    }
+
+    if (reaction.message.partial) {
+        try {
+            await reaction.message.member!.fetch();
+        } catch (error) {
+            log.error('Something went wrong when fetching the message: ', error);
             return;
         }
     }
 
-    const reactionsCount = await getTotalReactionCount(reaction)
+    if (reaction.message.member!.partial) {
+        try {
+            await reaction.message.member!.fetch();
+        } catch (error) {
+            log.error('Something went wrong when fetching the member: ', error);
+            return;
+        }
+    }
 
-    // if message owner gets 5+ reactions add "high value" role
-    if (reactionsCount >= REACTIONS_COUNT) {
-        const role = reaction.message.guild!.roles.cache.find((r) => r.name === ROLE.HIGH_VALUE.name);
-        const member = reaction.message.member;
-        await member!.roles.add(role!);
+    // ignore reactions from the message author
+    if (user.id === reaction.message.author.id) {
```

Comment:
I don't think this condition is enough, i can react to my message 4 times at least and then i will need 1 reaction from any other member to get the `high value` role.

Reply:
Oh you are right 👍 - I need to remove it from the count 🤦‍♂️ , great spot 👍

Reply:
I think this is now fixed 🤓